### PR TITLE
Add support php code sniffer json format

### PR DESCRIPTION
**Pull request by temp**
Add support for php code sniffer json format.

@DaveLiddament Got any better idea how to deal with the closures in PhpCodeSnifferJsonResultsParser::convertToString()? I need information from the full details, and didn't find anything else...

### CODE REVIEW

**Comment by DaveLiddament** [comment]
Wow, thanks very much for this PR. Looks fantastic. I'll review fully tomorrow  (Wednesday).

**Comment by DaveLiddament** [comment]
@temp Thanks very much for this. All looks great. There are a few tiny fixes, I'll add them in the review and then fix them.

File: src/Plugins/PhpCodeSnifferJsonResultsParser/PhpCodeSnifferJsonResultsParser.php [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\JsonParseException;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\JsonUtils;
+use DaveLiddament\StaticAnalysisResultsBaseliner\Domain\Utils\ParseAtLocationException;
+
+/**
+ * Handles PHPStan's JSON output.
+ *
+ * NOTE: SARB only deals with errors that are attached to a particular line in a file.
+ * PHPStan can report general errors (not specific to file). These are ignored by SARB.
+ */
```
Comment (DaveLiddament) [inline src/Plugins/PhpCodeSnifferJsonResultsParser/PhpCodeSnifferJsonResultsParser.php:38]: Update comment for phpcs. Or even delete:

```
/**
 * Handles PHP Code Sniffer's JSON output
 */
```

File: src/Plugins/PhpCodeSnifferJsonResultsParser/PhpCodeSnifferJsonResultsParser.php [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    private $fqcnRemover;
+
+    /**
+     * PhpstanJsonResultsParser constructor.
+     */
+    public function __construct(FqcnRemover $fqcnRemover)
+    {
+        $this->fqcnRemover = $fqcnRemover;
+    }
+
```
Comment (DaveLiddament) [inline src/Plugins/PhpCodeSnifferJsonResultsParser/PhpCodeSnifferJsonResultsParser.php:63]: This is not needed for PHP Code Sniffer as it give a 'type', in this case the source rule, for each violation it finds.

For more info about this option [this doc] (https://github.com/DaveLiddament/sarb/blob/master/docs/ViolationTypeClassificationGuessing.md), so far PHPStan is the only tool which needs it.

File: src/Plugins/PhpCodeSnifferJsonResultsParser/PhpCodeSnifferJsonResultsParser.php [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+     * @throws JsonParseException
+     */
+    private function convertAnalysisResultFromArray(
+        array $analysisResultAsArray,
+        FileName $fileName,
+        string $absoluteFilePath
+    ): AnalysisResult {
+        $lineAsInt = ArrayUtils::getIntValue($analysisResultAsArray, self::LINE);
+        $rawMessage = ArrayUtils::getStringValue($analysisResultAsArray, self::MESSAGE);
+        $rawType = ArrayUtils::getStringValue($analysisResultAsArray, self::TYPE);
```
Comment (DaveLiddament) [inline src/Plugins/PhpCodeSnifferJsonResultsParser/PhpCodeSnifferJsonResultsParser.php:239]: `$rawType` is not used, so safe to delete this line.

**Comment by temp** [comment]
Awesome! 👍

**Comment by temp** [comment]
@DaveLiddament Any chance to get a new release with this soon? ;-)

**Comment by DaveLiddament** [comment]
@temp sorry for the delay. 0.7.0 is ready: https://github.com/DaveLiddament/sarb/releases/tag/0.7.0

**Pull request merged**